### PR TITLE
Feat/dynamite/named-imports

### PR DESCRIPTION
**Pull request by Leptopoda**
Signed-off-by: Nikolas Rimikis <leptopoda@users.noreply.github.com>
currently based on #1400 (I was to lazy to rebase)
towards: #1399
also helps with: #1244 as it will only declare the import if it is referenced anywhere in the code.

remaining imports with explanation:
```dart
// din't touch dart imports yet.
dart:convert
dart:typed_data
// Typeresults need bigger refactors to make this work
package:built_collection/built_collection.dart
package:dynamite_runtime/models.dart
package:built_value/json_object.dart
// Needs bigger refactors but I have a branch from a while back
// when I already tried to tackle it
package:built_value/serializer.dart
// only exports extension methods so making it does not help
// Maybe I'll make it named anyways so we can't conflict with the extension name
package:collection/collection.dart
// Will never be possible as the built_value_generator would fail to run
package:built_value/built_value.dart
```

### CODE REVIEW

File: packages/dynamite/dynamite/lib/src/helpers/version_checker.dart
Comment (Leptopoda) [inline packages/dynamite/dynamite/lib/src/helpers/version_checker.dart:1]: Also needs to be removed from the readme

**Comment by Leptopoda** [comment]
@provokateurin should be ready now

**Review by provokateurin — APPROVED** [review]
This should completely close https://github.com/nextcloud/neon/issues/1399, right?

**Comment by Leptopoda** [comment]
Nope.

Some imports need bigger refractors and will come in the future

**Comment by provokateurin** [comment]
Ok but in the last commit you completely remove the old imports, that seems wrong if not all are migrated yet?

**Comment by Leptopoda** [comment]
I should have done a separate PR for that one sorry.
The last commit is just unused code. A leftover from before we generated a library for every spec.

Should I make a separate PR?

**Comment by provokateurin** [comment]
Yes please

**Pull request merged**